src/engine/Union.cpp

```diff
@@ -212,10 +229,19 @@ void Union::computeUnion(
       // The columns of the right subtree and the result match, we can
       // just copy the entries.
       if constexpr (RIGHT_WIDTH == OUT_WIDTH) {
-        res.insert(res.end(), right.begin(), right.end());
+        // always copy chunkSize results at once and then check for a timeout
+        size_t numChunks = right.size() / chunkSize;
+        for (size_t i = 0; i < numChunks; ++i) {
+          res.insert(res.end(), right.begin() + i * chunkSize,
+                     right.begin() + (i + 1) * chunkSize);
+          checkTimeout();
+        }
+        res.insert(res.end(), right.begin() + numChunks * chunkSize,
+                   right.end());
```

Comment:
Code duplication, in other places you have used a lambda to avoid this

Reply:
Indeed